src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -63,7 +64,9 @@ pub fn compile_targets<'a, 'cfg: 'a>(targets: &[(&'a Target, &'a Profile)],
                                      build_config: BuildConfig,
                                      profiles: &'a Profiles)
                                      -> CargoResult<Compilation<'cfg>> {
-    if targets.is_empty() {
+
+    let &(pkg, _) = pkg_targets.first().unwrap();
+    if pkg_targets.iter().any(|&(_, ref targets)| targets.is_empty()) {
```

Comment:
This isn't quite right any more I think because singling out the first package isn't really possible any more. This entire `if` may actually just be able to go away.